### PR TITLE
fix (some) SonarQube issues

### LINKED ISSUE #1799: issues raised by SonarQube static code analysis
resolves [SCSB-151](https://jira.stsci.edu/browse/SCSB-151)

before regression tests can be run on GHA for `drizzlepac`, these issues must be resolved:
https://plsonarqube.stsci.edu/project/issues?resolved=false&types=BUG&inNewCodePeriod=true&id=drizzlepac

- [x] `Change or remove this string; "getHelpAsString" is not defined.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/ablot.py#L32
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/findobj.py#L19-L23
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/pixreplace.py#L17

- [x] `Add 1 missing arguments; 'create_output' expects 2 positional arguments.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/adrizzle.py#L245
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/adrizzle.py#L312
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/adrizzle.py#L318

- [x] `Add 1 missing arguments; 'run' expects at least 2 positional arguments.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/devutils/comparison_tools/starmatch_hist.py#L870
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/starmatch_hist.py#L870

- [x] `Add 1 missing arguments; 'get_ci_info' expects at least 4 positional arguments.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/ci_table.py#L213

- [x] `Fix this attribute access on a value that can be 'None'.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/hapmultisequencer.py#L566
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/updatehdr.py#L520

- [x] `Remove this "return" statement from this "finally" block.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/hapsequencer.py#L758

- [x] `Verify this is the key that was intended; a value has already been saved for it on line 1437.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/align_utils.py#L1437-L1438

- [x] `Verify this is the key that was intended; a value has already been saved for it on line 517.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/tweakback.py#L521

- [x] `Fix this formatted string's syntax.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/astrometric_utils.py#L609

- [x] `Introduce a new variable or use its initial value before reassigning 'dao_threshold'.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/astrometric_utils.py#L960-L961

- [x] `Introduce a new variable or use its initial value before reassigning 'align_table'.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/product.py#L188

- [x] `Introduce a new variable or use its initial value before reassigning 'ivmlist'.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/processInput.py#L670

- [x] `Correct one of the identical sub-expressions on both sides of operator "and".`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/config_utils.py#L62

- [x] `str_ra is used before it is defined. Move the definition before.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/hapcut_utils.py#L118

- [x] `Remove or correct this useless self-assignment.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/mvm_quality_analysis.py#L892
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/haputils/product.py#L1522

- [x] `Raise this exception or remove this useless statement.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/imgclasses.py#L1052-L1053

- [x] `tddfile is used before it is defined. Move the definition before.`
https://github.com/spacetelescope/drizzlepac/blob/8fc110f831323f22c8af6b348e7dd3f081715ced/drizzlepac/nicmosData.py#L426

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-151](https://jira.stsci.edu/browse/SCSB-151)

Closes #1799

This PR addresses issues raised by SonarQube at https://plsonarqube.stsci.edu/project/issues?resolved=false&types=BUG&inNewCodePeriod=true&id=drizzlepac

**Checklist for maintainers**
- [ ] ~added entry in `CHANGELOG.rst` within the relevant release section~
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Comment by zacharyburnett** [comment]
I'm not sure if 77a347ac3a8b4646b2e748315cd2a904b041a5c6 and 6efbc7d1c4edf7c8579939d8a9af059acbc1aead are correct; 

- for 77a347ac3a8b4646b2e748315cd2a904b041a5c6 I used `outname` as the `filename` parameter 

- for 6efbc7d1c4edf7c8579939d8a9af059acbc1aead I added `log_level=0`, but I'm unsure if this is a good value

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
1. The starmatch_hist.py exists in two different directories. In order to remove one of them, more careful analysis has to be done.  Apparently this is fine for the moment, so please write a ticket for this so it can be fixed properly.
2. The default for log_level as a parameter for "run" in starmatch_hist.py should be the value of 20.

I am investigating the create_output() issue.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
The configObj['outdata' | 'outweight' | 'outcontext'] are the filenames and not numerical data.  The calls to **create_output()** should be 

create_output(configObj['outdata'], outsci)
create_output(configObj['outweight'], outwht)
create_output(configObj['outcontext'], outcon)

where the outXXX are computed on Line 237.  I see where the ".data" is then set to "outsci" (for example) in the main routine.  The files are opened in mode = "update" in **create_output()** so even though the extension is populated in **create_output()**, it is then overwritten with the same data (or a slightly modified version) in the main.  

Unfortunately, I cannot run/use TEAL, and this routine is not used when invoking "runastrodriz".

File: drizzlepac/adrizzle.py [hunk 1]
```diff
@@ -309,13 +309,13 @@ def run(configObj, wcsmap=None):
     out_sci_handle.close()
 
     if not util.is_blank(configObj['outweight']):
-        out_wht_handle, outwhtext = create_output(configObj['outweight'])
+        out_wht_handle, outwhtext = create_output(configObj['outweight'] outwht)
```
Comment (mdlpstsci) [inline drizzlepac/adrizzle.py:312]: For the want of a comma ...
Comma is missing between **configObj['outweight']** and **outwht**.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
It looks like adding the missing comma may be the only issue causing the failures.

File: drizzlepac/adrizzle.py (hunk 1 above)
Comment (zacharyburnett) [reply drizzlepac/adrizzle.py:312]: good catch!

**Review by mdlpstsci — APPROVED** [review]

**Pull request merged**